### ISSUE TITLE
Error with folders containing folders

### ISSUE THREAD

**Issue by 0xClandestine** [0xKitsune/solstat#19]
reproduce error:

1) forge init a blank repo

2) cd repo ; solstat -p src
... should work

3) create random folder in src ; solstat -p src
... should not work

**Comment by 0xClandestine** [comment]
Error in question:

```bash
thread 'main' panicked at 'Unable to read file: Os { code: 21, kind: IsADirectory, message: "Is a directory" }', /home/dev/.cargo/registry/src/github.com-1ecc6299db9ec823/solstat-0.2.0/src/analyzer/vulnerabilities/mod.rs:51:60
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Comment by 0xKitsune** [comment]
Thanks for pointing this out. Right now the program is set up to read files but not to read nested directories. 

This should be an easy fix though, I'll patch this one as well. Thanks again for opening an issue.

**Comment by 0xClandestine** [comment]
> Thanks for pointing this out. Right now the program is set up to read files but not to read nested directories.
> 
> This should be an easy fix though, I'll patch this one as well. Thanks again for opening an issue.

You're a 🐐, thanks again :)

**Comment by 0xKitsune** [comment]
Patched with [a2f63e609a52b81f07129497d23d80720c673fdd](https://github.com/0xKitsune/solstat/commit/a2f63e609a52b81f07129497d23d80720c673fdd)
